packages/database/src/core/view/ViewCache.ts

```diff
@@ -23,51 +23,54 @@ import { Node } from '../snap/Node';
  *
  * serverSnap is the cached server data, eventSnap is the cached event data (server data plus any local writes).
  */
-export class ViewCache {
-  constructor(
-    private readonly eventCache_: CacheNode,
-    private readonly serverCache_: CacheNode
-  ) {}
-
-  updateEventSnap(
-    eventSnap: Node,
-    complete: boolean,
-    filtered: boolean
-  ): ViewCache {
-    return new ViewCache(
-      new CacheNode(eventSnap, complete, filtered),
-      this.serverCache_
-    );
-  }
+export interface ViewCache {
+  readonly eventCache: CacheNode;
+  readonly serverCache: CacheNode;
+}
 
-  updateServerSnap(
-    serverSnap: Node,
-    complete: boolean,
-    filtered: boolean
-  ): ViewCache {
-    return new ViewCache(
-      this.eventCache_,
-      new CacheNode(serverSnap, complete, filtered)
-    );
-  }
+export function newViewCache(
```

Comment:
Nit: How about `createViewCache()`, which sounds more natural for a function?

Reply:
`new` is the pattern I have been using both in Firestore and in Database.